DEHPCommon/CommonUserInterface/ViewModels/Common/LoginViewModel.cs

```diff
@@ -0,0 +1,432 @@
+                    !string.IsNullOrEmpty(serverType.Value) && !string.IsNullOrEmpty(username) && !string.IsNullOrEmpty(password) &&
+                    !string.IsNullOrEmpty(uri));
+
+            this.WhenAnyValue(vm => vm.LoginFailed).Subscribe((loginFailed) =>
+            {
+                if (!loginFailed)
+                {
+                    return;
+                }
+                this.LogMessage($"Cannot login to {this.Uri}({this.ServerType.Value}) data-source");
+            });
+
+            this.WhenAnyValue(vm => vm.LoginSuccessfully).Subscribe(loginSuccessfully =>
+            {
+                if (!loginSuccessfully)
+                {
+                    return;
+                }
+                this.LogMessage($"Succesfully logged to {this.Uri}({this.ServerType.Value}) data-source");
+            });
+
+            this.WhenAnyValue(vm => vm.ServerType).Subscribe(_ =>
+            {
+                this.JsonIsSelected = this.ServerType.Key != null && this.ServerType.Key.Equals("JSON");
+            });
+
+            this.LoginCommand = ReactiveCommand.CreateAsyncTask(canLogin, x => this.ExecuteLogin(), RxApp.MainThreadScheduler);
+            this.LoadSourceFile = ReactiveCommand.Create();
+            this.LoadSourceFile.Subscribe(_ => this.ExecuteLoadSourceFile());
+            this.CheckUncheckModel = ReactiveCommand.Create();
+            this.CheckUncheckModel.Subscribe(_ => this.ExecuteCheckUncheckModel());
+
+            this.LoginSuccessfully = false;
+            this.LoginFailed = false;
+            this.EngineeringModels = new ReactiveList<EngineeringModelRowViewModel>();
+            this.EngineeringModels.ChangeTrackingEnabled = true;
+            this.SiteReferenceDataLibraries = new ReactiveList<SiteReferenceDataLibraryRowViewModel>();
+            this.SiteReferenceDataLibraries.ChangeTrackingEnabled = true;
+        }
+
+        /// <summary>
+        /// Executes login command
+        /// </summary>
+        /// <returns>The <see cref="Task"/></returns>
+        private async Task ExecuteLogin()
+        {
+            this.LoginSuccessfully = false;
+            this.LoginFailed = false;
+
+            try
+            {
+                if (this.IsSessionOpen(this.Uri, this.UserName, this.Password))
+                {
+                    this.LogMessage("The user is already logged on this server. Closing the session.");
+                    await this.ServerSession.Close();
+                }
+                var credentials = new Credentials(this.UserName, this.Password, new Uri(this.Uri));
+
+                switch (this.ServerType.Key)
+                {
+                    case "CDP":
+                        this.dal = new CdpServicesDal();
+                        break;
+                    case "OCDT":
+                        this.dal = new WspDal();
+                        break;
+                }
+
+                this.ServerSession = new Session(this.dal, credentials);
```

Comment:
The session will have to be assigned to the global session of the tool that will use this. SO this function should return it or somehow set a provided one.